File: PZ-Switch/src/zombie/iso/SearchMode.cpp
#include "zombie/iso/SearchMode.h"

namespace zombie {
namespace iso {

SearchMode SearchMode::getInstance() {
 // TODO: Implement getInstance
 return nullptr;
}

private
SearchMode::SearchMode() {
 // TODO: Implement SearchMode
 return nullptr;
}

float SearchMode::getFadeTime() {
 // TODO: Implement getFadeTime
 return 0;
}

void SearchMode::setFadeTime(float _fadeTime) {
 // TODO: Implement setFadeTime
}

bool SearchMode::isOverride(int plrIdx) {
 // TODO: Implement isOverride
 return false;
}

void SearchMode::setOverride(int plrIdx, bool enabled) {
 // TODO: Implement setOverride
}

bool SearchMode::isEnabled(int plrIdx) {
 // TODO: Implement isEnabled
 return false;
}

void SearchMode::setEnabled(int plrIdx, bool b) {
 // TODO: Implement setEnabled
}

void SearchMode::FadeIn(int int0) {
 // TODO: Implement FadeIn
}

void SearchMode::FadeOut(int int0) {
 // TODO: Implement FadeOut
}

void SearchMode::update() {
 // TODO: Implement update
}

void SearchMode::reset() {
 // TODO: Implement reset
}

public
SearchMode::PlayerSearchMode(int index, SearchMode sm) {
 // TODO: Implement PlayerSearchMode
 return nullptr;
}

bool SearchMode::isShaderEnabled() {
 // TODO: Implement isShaderEnabled
 return false;
}

bool SearchMode::isPlayerExterior() {
 // TODO: Implement isPlayerExterior
 return false;
}

float SearchMode::getShaderBlur() {
 // TODO: Implement getShaderBlur
 return 0;
}

float SearchMode::getShaderDesat() {
 // TODO: Implement getShaderDesat
 return 0;
}

float SearchMode::getShaderRadius() {
 // TODO: Implement getShaderRadius
 return 0;
}

float SearchMode::getShaderGradientWidth() {
 // TODO: Implement getShaderGradientWidth
 return 0;
}

float SearchMode::getShaderDarkness() {
 // TODO: Implement getShaderDarkness
 return 0;
}

void SearchMode::update() {
 // TODO: Implement update
}

private
SearchMode::SearchModeFloat(float float0, float float1, float float2) {
 // TODO: Implement SearchModeFloat
 return nullptr;
}

void SearchMode::set(float _exterior, float _targetExterior, float _interior,
 float _targetInterior) {
 // TODO: Implement set
}

void SearchMode::setTargets(float _targetExterior, float _targetInterior) {
 // TODO: Implement setTargets
}

float SearchMode::getExterior() {
 // TODO: Implement getExterior
 return 0;
}

void SearchMode::setExterior(float _exterior) {
 // TODO: Implement setExterior
}

float SearchMode::getTargetExterior() {
 // TODO: Implement getTargetExterior
 return 0;
}

void SearchMode::setTargetExterior(float _targetExterior) {
 // TODO: Implement setTargetExterior
}

float SearchMode::getInterior() {
 // TODO: Implement getInterior
 return 0;
}

void SearchMode::setInterior(float _interior) {
 // TODO: Implement setInterior
}

float SearchMode::getTargetInterior() {
 // TODO: Implement getTargetInterior
 return 0;
}

void SearchMode::setTargetInterior(float _targetInterior) {
 // TODO: Implement setTargetInterior
}

void SearchMode::update(float delta) {
 // TODO: Implement update
}

void SearchMode::equalise() {
 // TODO: Implement equalise
}

void SearchMode::reset() {
 // TODO: Implement reset
}

float SearchMode::getMin() {
 // TODO: Implement getMin
 return 0;
}

float SearchMode::getMax() {
 // TODO: Implement getMax
 return 0;
}

float SearchMode::getStepsize() {
 // TODO: Implement getStepsize
 return 0;
}

} // namespace iso
} // namespace zombie

File: converted/source2/source/zombie/iso/SearchMode.java
package zombie.iso;

import zombie.iso.SearchMode.PlayerSearchMode;
import zombie.iso.SearchMode.SearchModeFloat;

public class SearchMode {
   private static SearchMode instance;
   private float fadeTime = 1.0F;
   private PlayerSearchMode[] plrModes = new PlayerSearchMode[4];

   public static SearchMode getInstance() {
      if (instance == null) {
         instance = new SearchMode();
      }

      return instance;
   }

   private SearchMode() {
      for (int var1 = 0; var1 < this.plrModes.length; var1++) {
         this.plrModes[var1] = new PlayerSearchMode(var1, this);
         this.plrModes[var1].blur.setTargets(1.0F, 1.0F);
         this.plrModes[var1].desat.setTargets(0.85F, 0.85F);
         this.plrModes[var1].radius.setTargets(4.0F, 4.0F);
         this.plrModes[var1].darkness.setTargets(0.0F, 0.0F);
         this.plrModes[var1].gradientWidth.setTargets(4.0F, 4.0F);
      }
   }

   public PlayerSearchMode getSearchModeForPlayer(int var1) {
      return this.plrModes[var1];
   }

   public float getFadeTime() {
      return this.fadeTime;
   }

   public void setFadeTime(float var1) {
      this.fadeTime = var1;
   }

   public boolean isOverride(int var1) {
      return this.plrModes[var1].override;
   }

   public void setOverride(int var1, boolean var2) {
      this.plrModes[var1].override = var2;
   }

   public SearchModeFloat getRadius(int var1) {
      return this.plrModes[var1].radius;
   }

   public SearchModeFloat getGradientWidth(int var1) {
      return this.plrModes[var1].gradientWidth;
   }

   public SearchModeFloat getBlur(int var1) {
      return this.plrModes[var1].blur;
   }

   public SearchModeFloat getDesat(int var1) {
      return this.plrModes[var1].desat;
   }

   public SearchModeFloat getDarkness(int var1) {
      return this.plrModes[var1].darkness;
   }

   public boolean isEnabled(int var1) {
      return this.plrModes[var1].enabled;
   }

   public void setEnabled(int var1, boolean var2) {
      PlayerSearchMode var3 = this.plrModes[var1];
      if (var2 && !var3.enabled) {
         var3.enabled = true;
         this.FadeIn(var1);
      } else if (!var2 && var3.enabled) {
         var3.enabled = false;
         this.FadeOut(var1);
      }
   }

   private void FadeIn(int var1) {
      PlayerSearchMode var2 = this.plrModes[var1];
      var2.timer = Math.max(var2.timer, 0.0F);
      var2.doFadeIn = true;
      var2.doFadeOut = false;
   }

   private void FadeOut(int var1) {
      PlayerSearchMode var2 = this.plrModes[var1];
      var2.timer = Math.min(var2.timer, this.fadeTime);
      var2.doFadeIn = false;
      var2.doFadeOut = true;
   }

   public void update() {
      for (int var1 = 0; var1 < this.plrModes.length; var1++) {
         PlayerSearchMode var2 = this.plrModes[var1];
         var2.update();
      }
   }

   public static void reset() {
      instance = null;
   }
}
